src/main/java/handler/SupplyOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package handler;
+
+import java.util.Map;
+import model.FruitTransaction;
+
+public class SupplyOperationHandler implements OperationHandler {
+    private final Map<String, Integer> storage;
+
+    public SupplyOperationHandler(Map<String, Integer> storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        storage.put(transaction.getFruit(),
+                storage.getOrDefault(transaction.getFruit(), 0) + transaction.getQuantity());
+    }
```

Comment:
The class 'SupplyOperationHandler' should be placed in a package that groups it with other related classes. According to the checklist, you should have a 'strategy' package for holding handlers for different operations.